trin-storage/src/sql.rs

```diff
@@ -0,0 +1,62 @@
+// SQLite Statements
+
+pub const CREATE_QUERY: &str = "CREATE TABLE IF NOT EXISTS content_data (
+                                content_id_long TEXT PRIMARY KEY,
+                                content_id_short INTEGER NOT NULL,
+                                content_key TEXT NOT NULL,
+                                content_value TEXT NOT NULL,
+                                network INTEGER NOT NULL DEFAULT 0,
+                                content_size INTEGER
+                            );
+                            CREATE INDEX content_size_idx ON content_data(content_size);
+                            CREATE INDEX content_id_short_idx ON content_data(content_id_short);
+                            CREATE INDEX content_id_long_idx ON content_data(content_id_long);
+                            CREATE INDEX network_idx ON content_data(network);";
+
+pub const INSERT_QUERY: &str =
+    "INSERT OR IGNORE INTO content_data (content_id_long, content_id_short, content_key, content_value, network, content_size)
+                            VALUES (?1, ?2, ?3, ?4, ?5, ?6)";
+
+pub const INSERT_LC_UPDATE_QUERY: &str =
+    "INSERT OR IGNORE INTO lc_update (period, value, score, update_size)
+                            VALUES (?1, ?2, ?3, ?4)";
+
+pub const DELETE_QUERY: &str = "DELETE FROM content_data
+                            WHERE content_id_long = (?1)";
+
+pub const XOR_FIND_FARTHEST_QUERY: &str = "SELECT
+                                    content_id_long
+                                    FROM content_data
+                                    WHERE network = (?2)
+                                    ORDER BY ((?1 | content_id_short) - (?1 & content_id_short)) DESC";
+
+pub const CONTENT_KEY_LOOKUP_QUERY: &str =
+    "SELECT content_key FROM content_data WHERE content_id_long = (?1) LIMIT 1";
+
+pub const CONTENT_VALUE_LOOKUP_QUERY: &str =
+    "SELECT content_value FROM content_data WHERE content_id_long = (?1) LIMIT 1";
+
+pub const TOTAL_DATA_SIZE_QUERY: &str = "SELECT TOTAL(content_size) FROM content_data";
+
+pub const TOTAL_ENTRY_COUNT_QUERY: &str =
```

Comment:
Here's an example where I feel like naming could be improved. We have `TOTAL_DATA_SIZE_QUERY` which operates on all data in the db, and `TOTAL_ENTRY_COUNT_QUERY` which only operates on a subsection of data. From the `XStorage` level, it's impossible to distinguish b/w what is a network-specific/network-agnostic sql command without looking up the actual sql query. I think it would be beneficial if we introduce some kind of naming convention that easily distinguishes between those two types of sql queries....

Though, I'm struggling to come up with a good example :sweat_smile: - maybe just simple appending `DB` / `NETWORK` to the end of these would be sufficient? (at least for all the `content_data` table queries`)
eg...
- `TOTAL_ENTRY_COUNT_QUERY_NETWORK`
- `TOTAL_DATA_SIZE_QUERY_DB`
- `CONTENT_KEY_LOOKUP_QUERY_DB`
- `XOR_FIND_FARTHEST_QUERY_NETWORK`
- `PAGINATE_QUERY_DB` 
- etc...

Idk, if you feel like I'm being too nitpicky with this, I'm happy to back down. My sense is that if we're operating on a single db at the `XStorage` then it's pretty crucial to implicitly understand the effects of the sql queries when writing/reviewing/debugging `XStorage` level logic.

Reply:
We can indeed improve those, I will add the suffix DB/NETWORK for now.

Another way to abstract this, as you suggested, is to create some structs with adequate naming that will be responsible for the following:

- operates on the storage as a whole  - `StorageManager`
- operates on the network level of the storage `NetworkStorageManager`

Then we can add methods to those structs which will call the SQL queries. This way, any `*Storage` instances will not directly touch the SQL queries but will operate via the structs above and it will be more clear what is happening.